### PR TITLE
Argo code injection rule

### PR DESCRIPTION
**Pull request by 0xDC0DE**

### CODE REVIEW

File: yaml/argo/security/argo-workflow-parameter-command-injection.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
+rules:
+  - id: argo-workflow-parameter-command-injection
+    message: Using input or workflow parameters in here-scripts can lead to command injection or code injection. Convert the parameters to env variables instead.
+    languages: [yaml]
+    metadata:
+      category: security
+      cwe:
```
Comment (p4p3r) [inline yaml/argo/security/argo-workflow-parameter-command-injection.yaml:7]: You need to add an owasp reference to injection.

File: yaml/argo/security/argo-workflow-parameter-command-injection.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            command:
+              ...
+              - $LANG
+              ...
+            ...
+            source: 
+              $SCRIPT
+        - metavariable-regex:
+            metavariable: $LANG
+            regex: (bash|sh)
```
Comment (p4p3r) [inline yaml/argo/security/argo-workflow-parameter-command-injection.yaml]: I would use a regex more similar to the one here https://github.com/returntocorp/semgrep-rules/blob/develop/python/lang/security/dangerous-subprocess-use.yaml#L114 to match the shell executable.

**Review by p4p3r — COMMENTED** [review]
I added a couple of comments.
You have linting failures too, make sure you merge develop back into your branch to check if the failures comes from your changes.

File: yaml/argo/security/argo-workflow-parameter-command-injection.test.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
+apiVersion: argoproj.io/v1alpha1
```
Comment (p4p3r) [inline yaml/argo/security/argo-workflow-parameter-command-injection.test.yaml:1]: is the `apiVersion` key mandatory? You could add an additional `pattern-inside` to match `argoproj.io` to make sure the rule will report findings only for Argo yaml files.

**Review by p4p3r — CHANGES_REQUESTED** [review]

**Review by inkz — APPROVED** [review]

**Review by p4p3r — APPROVED** [review]
LGTM

**Pull request merged**